examples_tests/50.RadixSort/SortAndScatter.comp

```diff
@@ -0,0 +1,111 @@
+#version 450
+
+#define _NBL_GLSL_WORKGROUP_SIZE_ (1 << 8)
+
+layout (local_size_x = _NBL_GLSL_WORKGROUP_SIZE_) in;
+
+layout (set = 0, binding = 0, std430) readonly buffer in_buffer
+{
+	uvec2 in_values[];
+};
+
+layout (set = 0, binding = 1, std430) writeonly buffer out_buffer
+{
+	uvec2 out_values[];
+};
+
+layout (set = 0, binding = 2, std430) readonly buffer histogram_buffer
+{
+	uint histogram[];
+};
+
+layout(push_constant) uniform pushConstants
+{
+    layout (offset = 0) uint shift;
+} u_push_constants;
+
+#include <nbl/builtin/glsl/workgroup/arithmetic.glsl>
+
+#define BITS_PER_PASS 4
+#define NUM_BUCKETS (1 << BITS_PER_PASS)
+
+#define scratch_shared _NBL_GLSL_SCRATCH_SHARED_DEFINED_
+#define WG_DIM _NBL_GLSL_WORKGROUP_SIZE_
+
+shared uvec2 pingpong[WG_DIM * 2]; // 4kB
+shared uint local_histogram[NUM_BUCKETS]; // 64 bytes
+shared uint global_histogram[NUM_BUCKETS]; // 64 bytes
+
+void main()
```

Comment:
shouldn't all this just be (without local sort optimization)
```glsl
out_values[prefixSummedOffsets[gl_GlobalInvocationID.x]] = in_values[gl_GlobalInvocationID.x];
```

I serriously dont understand why you overcomplicate your life so much..

it should all be
```cpp
perValuePrefixSumOfABallot(); // does the first 256-prefix sums by itself
for (auto i=1u; i<=log_WG_size(elementCount-1u); i++)
for (auto j=0u; j<RadixBucketCount; j++)
   prefixSumUpsweep(j);
   
// top of hierarchy pass
for (auto j=0u; j<RadixBucketCount; j++)
   prefixSumUpsweep(j);
   
for (auto i=log_WG_size(elementCount-1u); i>0u; i++)
for (auto j=0u; j<RadixBucketCount; j++)
   prefixSumDownsweep(j);
   
scatter(); // does the last downsweep and then proceeds to scatter sort   
```

Reply:
Probably because I was 1-to-1 mapping the implementation from an older paper to new hardware. However, the biggest culprit was my lack of knowledge about in-kernel stream compaction with ballots. Lesson learned.